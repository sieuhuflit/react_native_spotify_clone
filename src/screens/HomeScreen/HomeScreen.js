import React from 'react';
import PropTypes from 'prop-types';
import { View, ScrollView } from 'react-native';
import { globalStyles } from '../../constants';
import AlbumsHorizontal from '../../components/AlbumsHorizontal';

class HomeScreen extends React.Component {
  static propTypes = {
    fetchRecommendation: PropTypes.func.isRequired,
    listAlbums: PropTypes.array
  };

  static defaultProps = {
    listAlbums: []
  };

  async componentDidMount() {
    const { fetchRecommendation } = this.props;
    fetchRecommendation();
  }

  render() {
    const { listAlbums } = this.props;
    return (
      <React.Fragment>
        <ScrollView
          showsVerticalScrollIndicator={false}
          style={globalStyles.container}
        >
          <View style={globalStyles.spacer8} />
          <AlbumsHorizontal
            data={listAlbums}
            heading="Best album in US"
            tagline="The music most voted this month."
          />
        </ScrollView>
      </React.Fragment>
    );
  }
}

export default HomeScreen;
