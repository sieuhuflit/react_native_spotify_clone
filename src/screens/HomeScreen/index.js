import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import HomeScreen from './HomeScreen';
import { fetchRecommendation } from '../../data/home/actions';
import HomeSelectors from '../../data/home/selectors';

const mapStateToProps = state => ({
  listAlbums: HomeSelectors.getListAlbums(state)
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ fetchRecommendation }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);
