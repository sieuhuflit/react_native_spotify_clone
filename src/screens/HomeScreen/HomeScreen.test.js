import React from 'react';
import { shallow } from 'enzyme';
import HomeScreen from './HomeScreen';

const setup = propOverrides => {
  const props = Object.assign(
    {
      fetchRecommendation: jest.fn()
    },
    propOverrides
  );
  const wrapper = shallow(<HomeScreen {...props} />);
  return {
    props,
    wrapper
  };
};

describe('HomeScreen check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('HomeScreen check life cycle', () => {
  test('should componentDidMount work', () => {
    const { wrapper, props } = setup();
    wrapper.instance().componentDidMount();
    expect(props.fetchRecommendation).toBeCalled();
  });
});
