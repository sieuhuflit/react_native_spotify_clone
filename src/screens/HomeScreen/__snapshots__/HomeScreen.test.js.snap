// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen check snapshot should match snapshot 1`] = `ShallowWrapper {}`;
